### PR TITLE
Add test for filehash58 offline mode.

### PR DESCRIPTION
**Pull request by matejak**
This PR adds a test that tests expected pass and failiure of the `filehash58` probe when doing offline scanning.

### CODE REVIEW

File: tests/probes/filehash58/test_probes_filehash58.sh [hunk 1]
```diff
@@ -43,10 +43,65 @@ function test_probes_filehash58 {
...
+    local ret_val=0;
+    local DF="$srcdir/check_filehash_simple.xml"
+
+    absolute_probe_root=$(cd "$1" && pwd)
+
+    # oscap-chroot is not readily available during test run, so we use oscap + env var setting.
+    result_keyword=$(OSCAP_PROBE_ROOT="$absolute_probe_root" "$OSCAP" oval eval "$DF" | grep oval_test_has_hash | grep -o '\w*$')
+    
+    [ "$result_keyword" == "$2" ] && return 0
+    # vvv This is more a test error than a failiure or "warning" vvv
```
Comment (jan-cerny) [inline tests/probes/filehash58/test_probes_filehash58.sh]: failiure

File: tests/probes/filehash58/test_probes_filehash58.sh [hunk 2]
```diff
@@ -43,10 +43,65 @@ function test_probes_filehash58 {
...
+	local ret_val=0
+
+	mkdir -p fail
+	echo bar > fail/oval-test
+
+	test_probes_filehash58_chroot fail false
+	ret_val=$?
+	rm -rf fail
+
+	[ $ret_val -eq 0 ] && rm -f /tmp/$FUNCNAME.tmp
```
Comment (jan-cerny) [inline tests/probes/filehash58/test_probes_filehash58.sh]: Where is this directory created?

File: tests/probes/filehash58/test_probes_filehash58.sh (hunk 1 above)
Comment (matejak) [reply tests/probes/filehash58/test_probes_filehash58.sh]: Good catch, fixed by rebase.

File: tests/probes/filehash58/test_probes_filehash58.sh (hunk 2 above)
Comment (matejak) [reply tests/probes/filehash58/test_probes_filehash58.sh]: Another good catch, I was puzzled by the first function of this test suite, fixed by rebase.

**Comment by jan-cerny** [comment]
@matejak Thank you very much. However, now we will have to merge `maint-1.2` to master and then port this test to CTest.

File: tests/probes/filehash58/test_probes_filehash58.sh [hunk 3]
```diff
@@ -38,15 +38,69 @@ function test_probes_filehash58 {
...
 
     return $ret_val
 }
 
+
+# $1: The chroot directory
+function test_probes_filehash58_chroot {
+
+    probecheck "filehash58" || return 255
+    require "sha1sum" || return 255
```
Comment (mpreisler) [inline tests/probes/filehash58/test_probes_filehash58.sh]: `sha1sum` is not actually used in the script, so why do we skip it if it's not found?

**Pull request merged**